src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java

```diff
@@ -106,27 +113,56 @@ private Button createSaveButton() {
         Button save = new Button();
         save.setText("Save");
         save.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
-        save.setDisableOnClick(true);
         save.addClassName("shelfFormSaveButton");
         save.addClickListener(event -> validateOnSave());
         return save;
     }
 
+    private Button createDeleteButton() {
+        delete = new Button();
+        delete.setText("Delete shelf");
+        delete.addThemeVariants(ButtonVariant.LUMO_ERROR);
+        delete.addClickListener(event -> validateOnDelete());
+        return delete;
+    }
```

Comment:
May be we can add a test for this as well?

Reply:
BooksInShelfViewTest will have a deleteShelfOnDeleteButtonClick test built for this.

Reply:
What needs to be tested here? This is just creating the button.

Reply:
Yes and also the button has correct label and it it does validation on click whatever is expected?